File: 1205-DefangingAnIpAddress/1205-DefangingAnIpAddress.js
// Last updated: 3/7/2026, 10:02:29 PM
/**
 * @param {string} address
 * @return {string}
 */
var defangIPaddr = function(address) {
     address = address.split('')
    
    for(let i=0; i<address.length; i++){
        if("." == address[i]){
            address[i] = "[.]"
        }
    }
    address = address.join('');
    console.log(address);
    return address;
};

File: 1205-DefangingAnIpAddress/1205-DefangingAnIpAddress.cpp
// Last updated: 8/18/2026, 11:51:19 AM
class Solution {
public:
    string defangIPaddr(string address) {
        for(int i=0; i<address.size(); i++){
            if(address[i] == '.'){
                address.replace(i, 1, "[.]");
                i += 2;
            }
        }
        return address;
    }
};